mlflow/transformers/__init__.py

```diff
@@ -514,14 +514,37 @@ def save_model(
         mlflow_model.signature = infer_signature_from_llm_inference_task(
             llm_inference_task, signature
         )
+        # The model with LLM inference task should accept a standard dictionary format
+        # alone so the example should not be converted to pandas DataFrame
+        example_no_conversion = True
     elif signature is not None:
         mlflow_model.signature = signature
 
     if input_example is not None:
         input_example = format_input_example_for_special_cases(input_example, built_pipeline)
         _save_example(mlflow_model, input_example, str(path), example_no_conversion)
+
     if metadata is not None:
         mlflow_model.metadata = metadata
+
+    # Check task consistency between model metadata and task argument
+    #  NB: Using mlflow_model.metadata instead of passed metadata argument directly, because
```

Comment:
This is required because [this line](https://github.com/mlflow/mlflow/blob/master/mlflow/models/model.py#L619) doesn't pass `metadata` to `flavor.save_model()` call. We could propagate that, but it may cause catastrophic issue in other flavors and model serving, so I take the safer workaround here.

Reply:
This is the safer bet :)